ddtrace/propagation/http.py

```diff
@@ -839,7 +839,8 @@ def _resolve_contexts(contexts, styles_w_ctx, normalized_headers):
         links = []
         for context in contexts[1:]:
             style_w_ctx = styles_w_ctx[contexts.index(context)]
-            if context.trace_id != primary_context.trace_id:
+            # encoding expects at least trace_id and span_id
+            if context.span_id and context.trace_id and context.trace_id != primary_context.trace_id:
```

Comment:
Some propagators get span ids to zero while other default to None. Can we make this behavior consistent? 

This isn't a blocker for this change.

Reply:
Actually, I think datadog, b3, and b3multi all return None, while tracecontext won't generate a context if the span_id or trace_id is 0, so I think we're good.